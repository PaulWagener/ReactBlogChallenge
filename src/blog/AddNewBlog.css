.add div {
  display: flex;
  margin: 10px;
}

.add label {
  flex: 3;
  font-size: 14pt;
  margin: 1em;
}

.add input, .add textarea {
  font-size: 14pt;
  padding: 5px;
  flex: 7;
}

.add textarea {
  height: 600px;
}

.add button {
  font-size: 30pt;
  margin: 30px auto;
  display: block;
}

.add form {
  padding: 30px;
}