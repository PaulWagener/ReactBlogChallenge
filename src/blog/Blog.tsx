
/**
 * Model class for a blogpost entry
 */
class Blog {
    id: Number;
    title: string;
    posted: Date;
    tags: String[];
    content: String;
}

export default Blog;