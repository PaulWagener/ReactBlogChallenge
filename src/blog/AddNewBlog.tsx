import * as React from 'react';
import './AddNewBlog.css';
import Blog from './Blog';

interface AddNewBlogState {
  title: string;
  content: string;
  tags: string;
}

interface AddNewBlogProps {
  addBlogHandler: (blog: Blog) => void;
}

class AddNewBlog extends React.Component<AddNewBlogProps, AddNewBlogState> {

  constructor() {
    super();
    this.state = {
      title: '',
      content: '',
      tags: ''
    };
    this.handleSubmit = this.handleSubmit.bind(this);
    this.handleInputChange = this.handleInputChange.bind(this);
  }

  handleInputChange(event: React.FormEvent<HTMLInputElement | HTMLTextAreaElement>) {
    const name = event.currentTarget.name;

    const data = {};
    data[name] = event.currentTarget.value;
    this.setState(data);
  }

  handleSubmit(event: React.FormEvent<HTMLFormElement>) {
    let blog = new Blog();
    blog.id = Math.round(Math.random() * 100000);
    blog.title = this.state.title;
    blog.content = this.state.content;
    blog.tags = this.state.tags.split(' ');
    blog.posted = new Date();
    this.props.addBlogHandler(blog);

    this.setState({
      title: '',
      content: '',
      tags: ''
    });
    alert('Blog is toegevoegd!');
  }

  render() {
    return (
      <div className="add">
        <form onSubmit={this.handleSubmit} ref="form">
          <div>
            <label>Titel:</label>
            <input type="text" name="title" value={this.state.title} onChange={this.handleInputChange} ref="title"/>
          </div>

          <div>
            <label>Content:</label>
            <textarea name="content" value={this.state.content} onChange={this.handleInputChange} ref="content"/>
          </div>

          <div>
            <label>Tags:</label>
            <input
              name="tags"
              placeholder="tv media foo"
              value={this.state.tags}
              onChange={this.handleInputChange}
              ref="tags"
            />
          </div>

          <button type="submit" ref="submit">Add blog</button>
        </form>
      </div>
    );
  }
}

export default AddNewBlog;
