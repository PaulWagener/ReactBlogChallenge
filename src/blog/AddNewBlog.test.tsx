// Link.react-test.js
import * as React from 'react';
import * as TestUtils from 'react-dom/test-utils'
import AddNewBlog from './AddNewBlog';
import Blog from './Blog';

describe('<AddNewBlog />', () => {
  it('renders', () => {
    expect(TestUtils.createRenderer().render(
      <AddNewBlog addBlogHandler={() => {}} />
    )).toMatchSnapshot()
  });

  it('Calls addBlogHandler', () => {
    const mockAddBlogHandler = jest.fn();
    const component = TestUtils.renderIntoDocument(<AddNewBlog addBlogHandler={mockAddBlogHandler} />) as AddNewBlog;

    // Fill in form
    (component.refs.title as HTMLInputElement).value = 'title';
    TestUtils.Simulate.change(component.refs.title);
    (component.refs.content as HTMLTextAreaElement).value = 'content';
    TestUtils.Simulate.change(component.refs.content);
    (component.refs.tags as HTMLInputElement).value = 'tag1 tag2';
    TestUtils.Simulate.change(component.refs.tags);

    // Submit form
    TestUtils.Simulate.submit(component.refs.form);

    // Check call to addBlogHandler
    expect(mockAddBlogHandler).toBeCalled();
    const blog = mockAddBlogHandler.mock.calls[0][0] as Blog;
    expect(blog.title).toBe('title');
    expect(blog.content).toBe('content');
    expect(blog.tags).toEqual(['tag1', 'tag2']);
  });
});