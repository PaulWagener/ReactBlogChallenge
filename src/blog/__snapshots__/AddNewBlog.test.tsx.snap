// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddNewBlog /> renders 1`] = `
initialize {
  "0": Object {
    "attribs": Object {},
    "children": Array [
      Object {
        "attribs": Object {
          "class": "add",
        },
        "children": Array [
          Object {
            "attribs": Object {},
            "children": Array [
              Object {
                "attribs": Object {},
                "children": Array [
                  Object {
                    "attribs": Object {},
                    "children": Array [
                      Object {
                        "data": "Titel:",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "text",
                      },
                    ],
                    "name": "label",
                    "next": Object {
                      "attribs": Object {
                        "name": "title",
                        "type": "text",
                        "value": "",
                      },
                      "children": Array [],
                      "name": "input",
                      "next": null,
                      "parent": [Circular],
                      "prev": [Circular],
                      "type": "tag",
                    },
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                  },
                  Object {
                    "attribs": Object {
                      "name": "title",
                      "type": "text",
                      "value": "",
                    },
                    "children": Array [],
                    "name": "input",
                    "next": null,
                    "parent": [Circular],
                    "prev": Object {
                      "attribs": Object {},
                      "children": Array [
                        Object {
                          "data": "Titel:",
                          "next": null,
                          "parent": [Circular],
                          "prev": null,
                          "type": "text",
                        },
                      ],
                      "name": "label",
                      "next": [Circular],
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                    },
                    "type": "tag",
                  },
                ],
                "name": "div",
                "next": Object {
                  "attribs": Object {},
                  "children": Array [
                    Object {
                      "attribs": Object {},
                      "children": Array [
                        Object {
                          "data": "Content:",
                          "next": null,
                          "parent": [Circular],
                          "prev": null,
                          "type": "text",
                        },
                      ],
                      "name": "label",
                      "next": Object {
                        "attribs": Object {
                          "name": "content",
                        },
                        "children": Array [],
                        "name": "textarea",
                        "next": null,
                        "parent": [Circular],
                        "prev": [Circular],
                        "type": "tag",
                      },
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                    },
                    Object {
                      "attribs": Object {
                        "name": "content",
                      },
                      "children": Array [],
                      "name": "textarea",
                      "next": null,
                      "parent": [Circular],
                      "prev": Object {
                        "attribs": Object {},
                        "children": Array [
                          Object {
                            "data": "Content:",
                            "next": null,
                            "parent": [Circular],
                            "prev": null,
                            "type": "text",
                          },
                        ],
                        "name": "label",
                        "next": [Circular],
                        "parent": [Circular],
                        "prev": null,
                        "type": "tag",
                      },
                      "type": "tag",
                    },
                  ],
                  "name": "div",
                  "next": Object {
                    "attribs": Object {},
                    "children": Array [
                      Object {
                        "attribs": Object {},
                        "children": Array [
                          Object {
                            "data": "Tags:",
                            "next": null,
                            "parent": [Circular],
                            "prev": null,
                            "type": "text",
                          },
                        ],
                        "name": "label",
                        "next": Object {
                          "attribs": Object {
                            "name": "tags",
                            "placeholder": "tv media foo",
                            "value": "",
                          },
                          "children": Array [],
                          "name": "input",
                          "next": null,
                          "parent": [Circular],
                          "prev": [Circular],
                          "type": "tag",
                        },
                        "parent": [Circular],
                        "prev": null,
                        "type": "tag",
                      },
                      Object {
                        "attribs": Object {
                          "name": "tags",
                          "placeholder": "tv media foo",
                          "value": "",
                        },
                        "children": Array [],
                        "name": "input",
                        "next": null,
                        "parent": [Circular],
                        "prev": Object {
                          "attribs": Object {},
                          "children": Array [
                            Object {
                              "data": "Tags:",
                              "next": null,
                              "parent": [Circular],
                              "prev": null,
                              "type": "text",
                            },
                          ],
                          "name": "label",
                          "next": [Circular],
                          "parent": [Circular],
                          "prev": null,
                          "type": "tag",
                        },
                        "type": "tag",
                      },
                    ],
                    "name": "div",
                    "next": Object {
                      "attribs": Object {
                        "type": "submit",
                      },
                      "children": Array [
                        Object {
                          "data": "Add blog",
                          "next": null,
                          "parent": [Circular],
                          "prev": null,
                          "type": "text",
                        },
                      ],
                      "name": "button",
                      "next": null,
                      "parent": [Circular],
                      "prev": [Circular],
                      "type": "tag",
                    },
                    "parent": [Circular],
                    "prev": [Circular],
                    "type": "tag",
                  },
                  "parent": [Circular],
                  "prev": [Circular],
                  "type": "tag",
                },
                "parent": [Circular],
                "prev": null,
                "type": "tag",
              },
              Object {
                "attribs": Object {},
                "children": Array [
                  Object {
                    "attribs": Object {},
                    "children": Array [
                      Object {
                        "data": "Content:",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "text",
                      },
                    ],
                    "name": "label",
                    "next": Object {
                      "attribs": Object {
                        "name": "content",
                      },
                      "children": Array [],
                      "name": "textarea",
                      "next": null,
                      "parent": [Circular],
                      "prev": [Circular],
                      "type": "tag",
                    },
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                  },
                  Object {
                    "attribs": Object {
                      "name": "content",
                    },
                    "children": Array [],
                    "name": "textarea",
                    "next": null,
                    "parent": [Circular],
                    "prev": Object {
                      "attribs": Object {},
                      "children": Array [
                        Object {
                          "data": "Content:",
                          "next": null,
                          "parent": [Circular],
                          "prev": null,
                          "type": "text",
                        },
                      ],
                      "name": "label",
                      "next": [Circular],
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                    },
                    "type": "tag",
                  },
                ],
                "name": "div",
                "next": Object {
                  "attribs": Object {},
                  "children": Array [
                    Object {
                      "attribs": Object {},
                      "children": Array [
                        Object {
                          "data": "Tags:",
                          "next": null,
                          "parent": [Circular],
                          "prev": null,
                          "type": "text",
                        },
                      ],
                      "name": "label",
                      "next": Object {
                        "attribs": Object {
                          "name": "tags",
                          "placeholder": "tv media foo",
                          "value": "",
                        },
                        "children": Array [],
                        "name": "input",
                        "next": null,
                        "parent": [Circular],
                        "prev": [Circular],
                        "type": "tag",
                      },
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                    },
                    Object {
                      "attribs": Object {
                        "name": "tags",
                        "placeholder": "tv media foo",
                        "value": "",
                      },
                      "children": Array [],
                      "name": "input",
                      "next": null,
                      "parent": [Circular],
                      "prev": Object {
                        "attribs": Object {},
                        "children": Array [
                          Object {
                            "data": "Tags:",
                            "next": null,
                            "parent": [Circular],
                            "prev": null,
                            "type": "text",
                          },
                        ],
                        "name": "label",
                        "next": [Circular],
                        "parent": [Circular],
                        "prev": null,
                        "type": "tag",
                      },
                      "type": "tag",
                    },
                  ],
                  "name": "div",
                  "next": Object {
                    "attribs": Object {
                      "type": "submit",
                    },
                    "children": Array [
                      Object {
                        "data": "Add blog",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "text",
                      },
                    ],
                    "name": "button",
                    "next": null,
                    "parent": [Circular],
                    "prev": [Circular],
                    "type": "tag",
                  },
                  "parent": [Circular],
                  "prev": [Circular],
                  "type": "tag",
                },
                "parent": [Circular],
                "prev": Object {
                  "attribs": Object {},
                  "children": Array [
                    Object {
                      "attribs": Object {},
                      "children": Array [
                        Object {
                          "data": "Titel:",
                          "next": null,
                          "parent": [Circular],
                          "prev": null,
                          "type": "text",
                        },
                      ],
                      "name": "label",
                      "next": Object {
                        "attribs": Object {
                          "name": "title",
                          "type": "text",
                          "value": "",
                        },
                        "children": Array [],
                        "name": "input",
                        "next": null,
                        "parent": [Circular],
                        "prev": [Circular],
                        "type": "tag",
                      },
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                    },
                    Object {
                      "attribs": Object {
                        "name": "title",
                        "type": "text",
                        "value": "",
                      },
                      "children": Array [],
                      "name": "input",
                      "next": null,
                      "parent": [Circular],
                      "prev": Object {
                        "attribs": Object {},
                        "children": Array [
                          Object {
                            "data": "Titel:",
                            "next": null,
                            "parent": [Circular],
                            "prev": null,
                            "type": "text",
                          },
                        ],
                        "name": "label",
                        "next": [Circular],
                        "parent": [Circular],
                        "prev": null,
                        "type": "tag",
                      },
                      "type": "tag",
                    },
                  ],
                  "name": "div",
                  "next": [Circular],
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                },
                "type": "tag",
              },
              Object {
                "attribs": Object {},
                "children": Array [
                  Object {
                    "attribs": Object {},
                    "children": Array [
                      Object {
                        "data": "Tags:",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "text",
                      },
                    ],
                    "name": "label",
                    "next": Object {
                      "attribs": Object {
                        "name": "tags",
                        "placeholder": "tv media foo",
                        "value": "",
                      },
                      "children": Array [],
                      "name": "input",
                      "next": null,
                      "parent": [Circular],
                      "prev": [Circular],
                      "type": "tag",
                    },
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                  },
                  Object {
                    "attribs": Object {
                      "name": "tags",
                      "placeholder": "tv media foo",
                      "value": "",
                    },
                    "children": Array [],
                    "name": "input",
                    "next": null,
                    "parent": [Circular],
                    "prev": Object {
                      "attribs": Object {},
                      "children": Array [
                        Object {
                          "data": "Tags:",
                          "next": null,
                          "parent": [Circular],
                          "prev": null,
                          "type": "text",
                        },
                      ],
                      "name": "label",
                      "next": [Circular],
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                    },
                    "type": "tag",
                  },
                ],
                "name": "div",
                "next": Object {
                  "attribs": Object {
                    "type": "submit",
                  },
                  "children": Array [
                    Object {
                      "data": "Add blog",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "button",
                  "next": null,
                  "parent": [Circular],
                  "prev": [Circular],
                  "type": "tag",
                },
                "parent": [Circular],
                "prev": Object {
                  "attribs": Object {},
                  "children": Array [
                    Object {
                      "attribs": Object {},
                      "children": Array [
                        Object {
                          "data": "Content:",
                          "next": null,
                          "parent": [Circular],
                          "prev": null,
                          "type": "text",
                        },
                      ],
                      "name": "label",
                      "next": Object {
                        "attribs": Object {
                          "name": "content",
                        },
                        "children": Array [],
                        "name": "textarea",
                        "next": null,
                        "parent": [Circular],
                        "prev": [Circular],
                        "type": "tag",
                      },
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                    },
                    Object {
                      "attribs": Object {
                        "name": "content",
                      },
                      "children": Array [],
                      "name": "textarea",
                      "next": null,
                      "parent": [Circular],
                      "prev": Object {
                        "attribs": Object {},
                        "children": Array [
                          Object {
                            "data": "Content:",
                            "next": null,
                            "parent": [Circular],
                            "prev": null,
                            "type": "text",
                          },
                        ],
                        "name": "label",
                        "next": [Circular],
                        "parent": [Circular],
                        "prev": null,
                        "type": "tag",
                      },
                      "type": "tag",
                    },
                  ],
                  "name": "div",
                  "next": [Circular],
                  "parent": [Circular],
                  "prev": Object {
                    "attribs": Object {},
                    "children": Array [
                      Object {
                        "attribs": Object {},
                        "children": Array [
                          Object {
                            "data": "Titel:",
                            "next": null,
                            "parent": [Circular],
                            "prev": null,
                            "type": "text",
                          },
                        ],
                        "name": "label",
                        "next": Object {
                          "attribs": Object {
                            "name": "title",
                            "type": "text",
                            "value": "",
                          },
                          "children": Array [],
                          "name": "input",
                          "next": null,
                          "parent": [Circular],
                          "prev": [Circular],
                          "type": "tag",
                        },
                        "parent": [Circular],
                        "prev": null,
                        "type": "tag",
                      },
                      Object {
                        "attribs": Object {
                          "name": "title",
                          "type": "text",
                          "value": "",
                        },
                        "children": Array [],
                        "name": "input",
                        "next": null,
                        "parent": [Circular],
                        "prev": Object {
                          "attribs": Object {},
                          "children": Array [
                            Object {
                              "data": "Titel:",
                              "next": null,
                              "parent": [Circular],
                              "prev": null,
                              "type": "text",
                            },
                          ],
                          "name": "label",
                          "next": [Circular],
                          "parent": [Circular],
                          "prev": null,
                          "type": "tag",
                        },
                        "type": "tag",
                      },
                    ],
                    "name": "div",
                    "next": [Circular],
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                  },
                  "type": "tag",
                },
                "type": "tag",
              },
              Object {
                "attribs": Object {
                  "type": "submit",
                },
                "children": Array [
                  Object {
                    "data": "Add blog",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "button",
                "next": null,
                "parent": [Circular],
                "prev": Object {
                  "attribs": Object {},
                  "children": Array [
                    Object {
                      "attribs": Object {},
                      "children": Array [
                        Object {
                          "data": "Tags:",
                          "next": null,
                          "parent": [Circular],
                          "prev": null,
                          "type": "text",
                        },
                      ],
                      "name": "label",
                      "next": Object {
                        "attribs": Object {
                          "name": "tags",
                          "placeholder": "tv media foo",
                          "value": "",
                        },
                        "children": Array [],
                        "name": "input",
                        "next": null,
                        "parent": [Circular],
                        "prev": [Circular],
                        "type": "tag",
                      },
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                    },
                    Object {
                      "attribs": Object {
                        "name": "tags",
                        "placeholder": "tv media foo",
                        "value": "",
                      },
                      "children": Array [],
                      "name": "input",
                      "next": null,
                      "parent": [Circular],
                      "prev": Object {
                        "attribs": Object {},
                        "children": Array [
                          Object {
                            "data": "Tags:",
                            "next": null,
                            "parent": [Circular],
                            "prev": null,
                            "type": "text",
                          },
                        ],
                        "name": "label",
                        "next": [Circular],
                        "parent": [Circular],
                        "prev": null,
                        "type": "tag",
                      },
                      "type": "tag",
                    },
                  ],
                  "name": "div",
                  "next": [Circular],
                  "parent": [Circular],
                  "prev": Object {
                    "attribs": Object {},
                    "children": Array [
                      Object {
                        "attribs": Object {},
                        "children": Array [
                          Object {
                            "data": "Content:",
                            "next": null,
                            "parent": [Circular],
                            "prev": null,
                            "type": "text",
                          },
                        ],
                        "name": "label",
                        "next": Object {
                          "attribs": Object {
                            "name": "content",
                          },
                          "children": Array [],
                          "name": "textarea",
                          "next": null,
                          "parent": [Circular],
                          "prev": [Circular],
                          "type": "tag",
                        },
                        "parent": [Circular],
                        "prev": null,
                        "type": "tag",
                      },
                      Object {
                        "attribs": Object {
                          "name": "content",
                        },
                        "children": Array [],
                        "name": "textarea",
                        "next": null,
                        "parent": [Circular],
                        "prev": Object {
                          "attribs": Object {},
                          "children": Array [
                            Object {
                              "data": "Content:",
                              "next": null,
                              "parent": [Circular],
                              "prev": null,
                              "type": "text",
                            },
                          ],
                          "name": "label",
                          "next": [Circular],
                          "parent": [Circular],
                          "prev": null,
                          "type": "tag",
                        },
                        "type": "tag",
                      },
                    ],
                    "name": "div",
                    "next": [Circular],
                    "parent": [Circular],
                    "prev": Object {
                      "attribs": Object {},
                      "children": Array [
                        Object {
                          "attribs": Object {},
                          "children": Array [
                            Object {
                              "data": "Titel:",
                              "next": null,
                              "parent": [Circular],
                              "prev": null,
                              "type": "text",
                            },
                          ],
                          "name": "label",
                          "next": Object {
                            "attribs": Object {
                              "name": "title",
                              "type": "text",
                              "value": "",
                            },
                            "children": Array [],
                            "name": "input",
                            "next": null,
                            "parent": [Circular],
                            "prev": [Circular],
                            "type": "tag",
                          },
                          "parent": [Circular],
                          "prev": null,
                          "type": "tag",
                        },
                        Object {
                          "attribs": Object {
                            "name": "title",
                            "type": "text",
                            "value": "",
                          },
                          "children": Array [],
                          "name": "input",
                          "next": null,
                          "parent": [Circular],
                          "prev": Object {
                            "attribs": Object {},
                            "children": Array [
                              Object {
                                "data": "Titel:",
                                "next": null,
                                "parent": [Circular],
                                "prev": null,
                                "type": "text",
                              },
                            ],
                            "name": "label",
                            "next": [Circular],
                            "parent": [Circular],
                            "prev": null,
                            "type": "tag",
                          },
                          "type": "tag",
                        },
                      ],
                      "name": "div",
                      "next": [Circular],
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                    },
                    "type": "tag",
                  },
                  "type": "tag",
                },
                "type": "tag",
              },
            ],
            "name": "form",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
          },
        ],
        "name": "div",
        "next": null,
        "parent": null,
        "prev": null,
        "root": [Circular],
        "type": "tag",
      },
    ],
    "name": "root",
    "next": null,
    "parent": null,
    "prev": null,
    "type": "root",
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xmlMode": false,
  },
}
`;
